snapshot/restore/local_fs_output.go

```diff
@@ -326,6 +336,81 @@ func write(targetPath string, r fs.Reader) error {
 	return nil
 }
 
+type sparseWriter struct {
+	f *os.File
+}
+
+// nolint:wrapcheck
+func (sw *sparseWriter) Write(b []byte) (int, error) {
+	w := 0
+
+	bSize, err := stat.GetFileAllocSize(sw.f.Name())
+	if err != nil {
+		return w, err // nolint:wrapcheck
+	}
+
+	for i, j := 0, bSize; i < len(b); i, j = int(j), j+bSize {
```

Comment:
i think this makes a hidden assumption that `Write()` is always called for multiples `bSize`. Instead of complicating writer we should just make a dedicated iocopy.CopySparse() .